public/app/plugins/datasource/loki/datasource.ts

```diff
@@ -102,6 +102,38 @@ export class LokiDatasource extends DataSourceApi<LokiQuery, LokiOptions> {
     return getBackendSrv().fetch<Record<string, any>>(req);
   }
 
+  getLogsVolumeQuery(request: DataQueryRequest<LokiQuery>): Observable<DataQueryResponse> | undefined {
+    const isLogsVolumeAvailable = request.targets.some((target) => !isMetricsQuery(target.expr) && !!target.expr);
+    if (!isLogsVolumeAvailable) {
+      return;
+    }
+
+    const histogramRequest = cloneDeep(request!);
+    histogramRequest.targets = histogramRequest.targets
+      .filter((target) => !isMetricsQuery(target.expr))
+      .map((target) => {
+        target.expr = `count_over_time(${target.expr}[$__interval])`;
+        return target;
+      });
+
+    return new Observable((observer) => {
+      const subscription = this.query(histogramRequest!)
```

Comment:
```suggestion
      const subscription = this.query(histogramRequest)
```

is the ! needed here? it seems to me that histogramRequest cannot be nullish.

Reply:
Fixed in #39042